### PR TITLE
Speed

### LINKED ISSUE #103: `interface.npy` remains in folder, should be removed after run (?)

### PR DESCRIPTION
**Pull request by Rykath**
* Fix #103 (clean `Runner` after `profit run`, clean `log` with `profit clean`, `log` relative to `base_dir`)
* Enhancement #100 (speedup of roughly x30)
  * modified `LocalRunner` to use `multiprocessing.Process` (uses `fork` internally) instead of a subprocess (via a shell)
  * added a new *custom* `Worker` similar to previous custom components which is directly loaded with python
  in combination with the new `LocalRunner` this means a short simulation written in python can be called directly (no external call like `python3 mockup.py`) with little overhead, without importing `profit` each run
* new testcases for the custom workers
* `example/mockup.py` enhanced to be able to process up to 5 input variables

### CODE REVIEW

File: profit/sur/gaussian_process.py [hunk 1]
```diff
@@ -5,6 +5,8 @@
 from .sur import Surrogate
 from profit.util import check_ndim
 
+from sklearn.gaussian_process.kernels import Kernel, Hyperparameter, StationaryKernelMixin, NormalizedKernelMixin
```
Comment (krystophny) [inline profit/sur/gaussian_process.py:8]: Why do we add this here? Is there no performance overhead?

**Review by krystophny — COMMENTED** [review]
@Rykath thanks for the nice work! Just one last question: Why are the sklearn things imported on top now?

**Comment by krystophny** [comment]
... Oh and I just see there were some issues when merging the other two PRs before. Can you try to rebase on `upstream/master` and resolve conflicts?

File: profit/sur/gaussian_process.py (hunk 1 above)
Comment (Rykath) [reply profit/sur/gaussian_process.py:8]: There is a significant performance overhead, but the previous location of the imports had exactly the same overhead, but less readability. I have not yet found an (easy) way around this import as it is needed in the class declaration.

**Comment by Rykath** [comment]
Regarding the performance due to imports: With the current structure avoiding imports is quite difficult as reading the config requires calling `handle_config` of various classes. One option would be to restructure the `run` and `sur` components to have separate `handle_config` functions at another location.
The other option (in part what we are doing already) is to avoid restarting a python instance and do as much as possible internally or by forking the current process. This of course does not work that well on a cluster.

**Comment by Rykath** [comment]
I’m quite dissatisfied with how we handle the imports in `sur/gaussian_process.py`. In my opinion readability is very bad and it does not really work well (e.g. `sklearn`).
There might be a way to solve the fundamental problem, that `run` and `sur` are imported from `config`. We could remove the respective `handle_config` calls from `config` as the respective config is usually only needed when the components are initialized. The `init` method could then call `handle_config` for example.
We should probably discuss this on friday.

**Review by krystophny — APPROVED** [review]

**Pull request merged**